controllers/host/factory.go

```diff
@@ -0,0 +1,53 @@
+package host
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	starlingxv1 "github.com/wind-river/cloud-platform-deployment-manager/api/v1"
+	"github.com/wind-river/cloud-platform-deployment-manager/controllers/common"
+)
+
+// During factory install, the reconciled status is expected to be updated to
+// false to unblock the configuration as the day 1 configuration.
+// updateStatusForFactoryInstall
+func (r *HostReconciler) updateStatusForFactoryInstall(ctx context.Context, obj client.Object) []reconcile.Request {
+	IgnoreReconcile := []reconcile.Request{}
+	namespace := r.GetNamespace()
+	if _, ok := common.FactoryConfigAllowed(namespace, obj); !ok {
+		return IgnoreReconcile
+	}
+
+	log := logHost.WithName("factory")
+	log.Info("starting config update")
+
+	hosts := &starlingxv1.HostList{}
+	opts := client.ListOptions{Namespace: namespace}
+
+	if err := r.Client.List(context.TODO(), hosts, &opts); err != nil {
+		log.Error(err, "failed to retrieve the resources")
+		return IgnoreReconcile
+	}
+
+	if len(hosts.Items) == 0 {
+		log.Info("not hosts found")
+		return IgnoreReconcile
+	}
+
+	for _, host := range hosts.Items {
+		host.Status.Reconciled = false
+		host.Status.Defaults = nil
```

Comment:
Please add a comment here to explain: set host defaults to nill for re-collecting the default post enrollment